eth-providers/src/base-provider.ts

```diff
@@ -1572,29 +1571,34 @@ export abstract class BaseProvider extends AbstractProvider {
     ]);
   };
 
-  _getTXReceipt = async (txHash: string): Promise<TransactionReceipt | TransactionReceiptGQL> => {
-    if (await this._isTXPending(txHash)) {
-      const txFromNextBlock = await this._getTXReceiptFromNextBlock(txHash);
-      if (txFromNextBlock) return txFromNextBlock;
-    }
+  _getTXReceipt = async (txHash: string): Promise<TransactionReceipt | TransactionReceiptGQL | null> => {
+    // TODO: potentially these 3 can go in parallel
+    let res = null;
+    try {
+      if (await this._isTXPending(txHash)) {
+        const txFromNextBlock = await this._getTXReceiptFromNextBlock(txHash);
+        if (txFromNextBlock) return txFromNextBlock;
+      }
 
-    const txFromCache = await this._getTxReceiptFromCache(txHash);
-    if (txFromCache) return txFromCache;
+      const txFromCache = await this._getTxReceiptFromCache(txHash);
+      if (txFromCache) return txFromCache;
 
-    try {
       const txFromSubql = await this.subql?.getTxReceiptByHash(txHash);
-      return txFromSubql || logger.throwError(`transaction hash not found`, Logger.errors.UNKNOWN_ERROR, { txHash });
+      res = txFromSubql || null;
     } catch {
-      return logger.throwError(`transaction hash not found`, Logger.errors.UNKNOWN_ERROR, { txHash });
+      res = null;
     }
+
+    return res;
   };
 
   // Queries
   getTransaction = (txHash: string): Promise<TransactionResponse> =>
     throwNotImplemented('getTransaction (deprecated: please use getTransactionByHash)');
 
-  getTransactionByHash = async (txHash: string): Promise<TX> => {
+  getTransactionByHash = async (txHash: string): Promise<TX | null> => {
     const tx = await this._getTXReceipt(txHash);
+    if (!tx) return null;
 
     const extrinsic = await this._getExtrinsicsAtBlock(tx.blockHash, txHash);
 
```

Comment:
```
const from = tx.from ||  await this.getEvmAddress(args.origin).toLowerCase();// not sure

return {
      from: from,
      to: args.target || null,
      hash: tx.transactionHash,
      blockHash: tx.blockHash || null,
      nonce,
      blockNumber: tx.blockNumber || null,
      transactionIndex: tx.transactionIndex || null,
      value,
      gasPrice: GAS_PRICE,
      gas: args.gas_limit,
      input
    };
```

Reply:
```
const tx = await this._getTXReceipt(txHash);
```
if tx is pending, this will call `_getTXReceiptFromNextBlock()`, so we will need some more refactor to make #125 work. So let's keep this PR in it's scope to fix `eth_getTransactionReceipt`